internal/models/sessions.go

```diff
@@ -104,6 +106,54 @@ func (s *Session) UpdateOnlyRefreshInfo(tx *storage.Connection) error {
 	return tx.UpdateOnly(s, "refreshed_at", "user_agent", "ip")
 }
 
+type SessionValidityReason = int
+
+const (
+	SessionValid        SessionValidityReason = iota
+	SessionPastNotAfter                       = iota
+	SessionPastTimebox                        = iota
+	SessionTimedOut                           = iota
+)
+
+func (s *Session) CheckValidity(now time.Time, refreshTokenTime *time.Time, timebox, inactivityTimeout *time.Duration) SessionValidityReason {
+	if s.NotAfter != nil && now.After(*s.NotAfter) {
+		return SessionPastNotAfter
+	}
+
+	if timebox != nil && *timebox != 0 && now.After(s.CreatedAt.Add(*timebox)) {
```

Comment:
don't we already check that the timebox can never by 0 when it's set in the config?

Reply:
We do for the config, but the config is not this function. I can very easily send a 0 here and there's nothing to stop me from making a mess.